cpp/src/Slice/Grammar.y

```diff
@@ -22,6 +22,9 @@
 #define YYMAXDEPTH  10000
 #define YYINITDEPTH YYMAXDEPTH
 
+// Newer bison versions allow to disable stack resizing by defining yyoverflow.
```

Comment:
While working on updating `IceStorm`, I saw `yyoverflow` and realized it was missing from the Slice grammar.
We had it on 4.0 (here it is on the 'before_streamline' tag: https://github.com/zeroc-ice/ice/blob/6e08b82a0298696c470137808425893455bc0e76/cpp/src/Slice/Grammar.y#L25-L26).

But I guess it got lost among the backporting, so I added it back. Doing so meant I also had to add another `warning disable` macro, because of how trivial this definition is (it just errors, so some code is now unused)

This function is necessary because we _cannot_ allow flex (the scanner) to reallocate a new stack.
Flex uses simple alloc/memcpy calls which don't work with our C++ types.
Flex supports a C++ mode, but we never switched to it.

Reply:
What about making the same update to the IceGrid grammar file?